src/neural/blas/network_blas.cc

```diff
@@ -310,16 +311,134 @@ void BlasComputation<use_eigen>::ComputeBlocking() {
           SELU,  // SELU activation for attention head.
           head_buffer.data());
 
-      for (auto layer : weights_.pol_encoder) {
-        // TODO: support encoder heads.
-        throw Exception(
-            "Eigen/Blas backend doesn't support encoder heads yet.");
-      }
       const size_t policy_d_model = weights_.ip2_pol_b.size();
-      std::vector<float> head_buffer2(largest_batch_size * policy_d_model *
+      const size_t max_channel_size = weights_.pol_encoder.size() > 0
+                            ? weights_.pol_encoder[0].ffn.dense1_b.size() // DFF size
+                            : policy_d_model;
+      std::vector<float> head_buffer2(largest_batch_size * max_channel_size *
+                                      kSquares);
+      std::vector<float> head_buffer3(largest_batch_size * max_channel_size *
                                       kSquares);
-      std::vector<float> head_buffer3(largest_batch_size * policy_d_model *
+      std::vector<float> head_buffer4(largest_batch_size * max_channel_size *
                                       kSquares);
+      std::vector<float> temp_buffer1(policy_d_model * kSquares);
```

Comment:
Can you move this block inside an if statement so that the allocations are only done if needed?